Tests/TensorFlowTests/LayerTests.swift

```diff
@@ -1935,6 +1935,58 @@ final class LayerTests: XCTestCase {
     XCTAssertEqual(transformerTensor.shape, transformerResult.shape)
   }
 
+  func testGroupNorm() {
```

Comment:
Could you also add gradient tests for these layers?

Reply:
Added.
The code got awful after `swift-format`😂

Reply:
@t-ae I'd still suggest manual refactoring tho xD